data/crac/crac-util/src/main/java/com/farao_community/farao/data/crac_util/CracCleaner.java

```diff
@@ -0,0 +1,131 @@
+                }
+            });
+            unmonitoredCnecs.forEach(cnec -> crac.removeCnec(cnec.getId()));
+        }
+
+        // remove RangeAction whose NetworkElement is absent from the network
+        ArrayList<RangeAction> absentFromNetworkRangeActions = new ArrayList<>();
+        for (RangeAction rangeAction: crac.getRangeActions()) {
+            rangeAction.getNetworkElements().forEach(networkElement -> {
+                if (network.getIdentifiable(networkElement.getId()) == null) {
+                    absentFromNetworkRangeActions.add(rangeAction);
+                    report.add(String.format("[REMOVED] Remedial Action %s with network element [%s] is not present in the network. It is removed from the Crac", rangeAction.getId(), networkElement.getId()));
+                }
+            });
+        }
+        absentFromNetworkRangeActions.forEach(rangeAction -> crac.removeRangeAction(rangeAction.getId()));
+
+        // remove NetworkAction whose NetworkElement is absent from the network
+        ArrayList<NetworkAction> absentFromNetworkNetworkActions = new ArrayList<>();
+        for (NetworkAction networkAction: crac.getNetworkActions()) {
+            networkAction.getNetworkElements().forEach(networkElement -> {
+                if (network.getIdentifiable(networkElement.getId()) == null) {
+                    absentFromNetworkNetworkActions.add(networkAction);
+                    report.add(String.format("[REMOVED] Remedial Action %s with network element [%s] is not present in the network. It is removed from the Crac", networkAction.getId(), networkElement.getId()));
+                }
+            });
+        }
+        absentFromNetworkNetworkActions.forEach(networkAction -> crac.removeNetworkAction(networkAction.getId()));
+
+        // remove Contingencies whose NetworkElement is absent from the network or does not fit a valid Powsybl Contingency
+        Set<Contingency> absentFromNetworkOrUnhandledContingencies = new HashSet<>();
+        for (Contingency contingency : crac.getContingencies()) {
+            contingency.getNetworkElements().forEach(networkElement -> {
+                Identifiable<?> identifiable = network.getIdentifiable(networkElement.getId());
+                if (identifiable == null) {
+                    absentFromNetworkOrUnhandledContingencies.add(contingency);
+                    report.add(String.format("[REMOVED] Contingency %s with network element [%s] is not present in the network. It is removed from the Crac", contingency.getId(), networkElement.getId()));
+                } else if (!(identifiable instanceof Branch || identifiable instanceof Generator || identifiable instanceof HvdcLine || identifiable instanceof BusbarSection || identifiable instanceof DanglingLine)) {
+                    if (!REMOVE_UNHANDLED_CONTINGENCIES.isEnabled()) {
```

Comment:
Invert the condition, as it will be easier to read

Reply:
Done